### DISCUSSION TITLE
Defining what ports listen for...

### DISCUSSION THREAD

**Discussion by glaklandy** [mido/mido#390] [Q&A] [answered]
Hello! First of all, so many thanks for **mido**! 🥇 

Might there be a way that I've overlooked to limit the messages e.g., types that ports listen for? Put another way, can ports be made to ignore certain messages?

For example, I have `portA` and `portB` as input ports, but I only want `portA` to listen for MIDI clock, start, and stop messages, ignoring all others, while I want `portB` to ignore clock, start, and stop, and explicitly only listen for notes, CCs, and PCs... 

I've been looking and thinking of a way to do this either as part of the port definition/creation or at least as part of the port monitoring (i.e., `for i in portA.iter_pending():`  etc.), but have only come up with basic solutions using if statements after this monitoring e.g., `if msg == mido.Message('start') :  #ignore `

Perhaps it doesn't really matter though and/or I'm thinking about it all wrong! Any suggestions/advice is welcome. :)

**Comment by rdoursenaud** [comment, marked as answer]
As far as I know, mido doesn't feature any filtering capabilities. You'll have to do the filtering yourself.

**Reply by glaklandy** [reply]
> As far as I know, mido doesn't feature any filtering capabilities. You'll have to do the filtering yourself.

Thanks for the response. This is how I understood it too, but just wanted to check there wasn't a better and/or built-in method. :)
